packages/program/src/functions/invocation-results.ts

```diff
@@ -73,6 +72,27 @@ export class InvocationResult<T = any> {
     return this.functionScopes[0].getCallConfig();
   }
 
+  /**
+   * Gets the ABI from all calls.
+   *
+   * @returns The ABIs from all calls.
+   */
+  getAbiFromAllCalls = () =>
+    this.functionScopes.reduce((acc, funcScope, i) => {
+      const { program, externalAbis } = funcScope.getCallConfig();
+
+      if (i === 0) {
```

Comment:
Can we always assume that the first ABI is the non external?

Reply:
Yes because the first function scope entry is always the instance that initiates the call scope

```ts
// main ABI belongs to contractA
contractA.functions.x_y_z(10).addContracts([contractB])

or

// main ABI belongs to contractB
contractB.multicall(...)

or

// main ABI belongs to script
scriptA.functions.main()
```

Since the main ABI can belong to either a script or contract, we cannot index it by an `id`. I am treating it differently from the other ABIs

The following ABIs will always belong to contracts since we can't execute a multi-call using a script invocation or even make a script to call another script.